Stride-Labs/stride
# add alliance module

## Context and purpose of the change
The Alliance module is a new Cosmos SDK-compatible module that allows assets from an established blockchain to be staked on a newer blockchain.
The main purpose of the change is to integrate Alliance module into stride so that users can stake stTokens.

## Brief Changelog
  - *Update app.go file for Alliance module integration*
  - *Add integration tests for alliance module*
 
## How to test using CLI
1. IBC transfer tokens from one chain (for example, gaia) to stride and do liquid stake
2. Submit proposal for registering stTokens as alliance asset.
Check the following example clis.
```
stridedl tx gov submit-legacy-proposal create-alliance stuatom 0.5 0.5 1 3600s --from=test1 -y
stridedl tx gov deposit 1 10000001ustrd --from val1 -y
stridedl tx gov vote 1 yes --from val1 -y
stridedl tx gov vote 1 yes --from val2 -y
stridedl tx gov vote 1 yes --from val3 -y
```
3. Test delegating/redelegating/undelegating/claiming features.
Check the following example clis.
```
stridedl tx alliance delegate stridevaloper1nnurja9zt97huqvsfuartetyjx63tc5zrj5x9f 100stuatom --from val1 -y
stridedl tx alliance redelegate stridevaloper1nnurja9zt97huqvsfuartetyjx63tc5zrj5x9f stridevaloper1uk4ze0x4nvh4fk0xm4jdud58eqn4yxhrgpwsqm 50stuatom --from val1 -y
stridedl tx alliance undelegate stridevaloper1nnurja9zt97huqvsfuartetyjx63tc5zrj5x9f 100stuatom --from val1 -y
stridedl tx alliance claim-rewards stridevaloper1nnurja9zt97huqvsfuartetyjx63tc5zrj5x9f stuatom --from val1 -y
```

Use [this link](https://alliance.terra.money/guides/how-to) to check all available tx, query CLIs. 

## Integration tests
Run the following command to run integration tests.
`make test-integration-docker`

Here is the screenshot of integration test result.

<img width="763" alt="Screenshot 2023-01-26 at 1 46 57 PM" src="https://user-images.githubusercontent.com/36045227/214923115-47675206-4d6a-4d6a-92ae-6385bf889cff.png">
